libs/designer/src/lib/ui/panel/connectionsPanel/createConnection/createConnectionWrapper.tsx

```diff
@@ -277,7 +279,7 @@ export const CreateConnectionInternal = (props: {
         let connection: Connection | undefined;
         let err: string | undefined;
 
-        const newName = await getUniqueConnectionName(connector.id, existingReferences);
+        const newName = connectionName ? connectionName : await getUniqueConnectionName(connector.id, existingReferences);
```

Comment:
Nit: We could just use `??` here 
```suggestion
        const newName = connectionName ?? await getUniqueConnectionName(connector.id, existingReferences);
```

Reply:
Will do this in my next PR